File: BFS of graph[31-July-2023]/Solution.py
#  >> PYTHON CODE <<

from typing import List
from queue import Queue

class Solution:
    #Function to return Breadth First Traversal of given graph.
    def bfsOfGraph(self, V: int, adj: List[List[int]]) -> List[int]:
        
        ans=[]
        q=Queue()
        visited=[False]*V
        
        q.put(0)
        visited[0]=True
        
        while(not q.empty()):
            u=q.get()
            
            ans.append(u)
            
            for el in adj[u]:
                if visited[el]==False:
                    q.put(el)
                    visited[el]=True

        return ans

File: BFS of graph[31-July-2023]/Solution.cpp
//  >> CPP CODE <<

class Solution {
  public:
    // Function to return Breadth First Traversal of given graph.
    vector<int> bfsOfGraph(int V, vector<int> adj[]) {
        vector<int> ans;
        queue<int> q;
        bool visited[V] = {false};
        
        q.push(0);
        visited[0] = true;
        
        while(!q.empty()) {
            int u = q.front();
            q.pop();
            
            ans.push_back(u);
            
            for(auto el : adj[u]) {
                if(!visited[el]) {
                    q.push(el);
                    visited[el] = true;
                }
            }
        }
        
        return ans;
    }
};